packages/astro/test/units/routing/manifest.test.js

```diff
@@ -148,6 +148,66 @@ describe('routing - createRouteManifest', () => {
 		]);
 	});
 
+	it('route sorting respects the file tree', async () => {
+		const fs = createFs(
+			{
+				'/src/pages/[dynamic]/static.astro': `<h1>test</h1>`,
+				'/src/pages/[dynamic]/index.astro': `<h1>test</h1>`,
+				'/src/pages/[dynamic]/[...rest].astro': `<h1>test</h1>`,
+				'/src/pages/[...rest]/static.astro': `<h1>test</h1>`,
+				'/src/pages/[...rest]/index.astro': `<h1>test</h1>`,
+				'/src/pages/static.astro': `<h1>test</h1>`,
```

Comment:
can we also add another route like `src/pages/blog/index.astro` and `src/pages/blog.astro`?

Reply:
We can't have both simultaneously since they would define the same thing. But I'll add the missing examples